packages/effect/src/Number.ts

```diff
@@ -493,3 +493,64 @@ export const parse = (s: string): Option<number> => {
     ? option.none
     : option.some(n)
 }
+
+/**
+ * Takes a string and returns an `Option` of `number`.
+ *
+ * If the string is empty or invalid characters,
+ * it returns `option.none`. Otherwise, it attempts to convert the string to a number
+ * and returns `option.some(number)`.
+ *
+ * @param s - The string to be converted to a `number`.
+ * @returns An `Option` type that is either `some(number)` if the string can be converted
+ *          to a valid number, or `none` if the string is empty or contains invalid characters
+ *          for number conversion.
+ *
+ * @example
+ * import { fromString } from 'effect/Number'
+ *
+ * assert.deepStrictEqual(fromString("42"), option.some(42))
+ * assert.deepStrictEqual(fromString(""), option.none)
+ * assert.deepStrictEqual(fromString("not a number"), option.none)
+ *
+ * @category conversions
+ * @since 2.4.12
+ */
+
+export const fromString = (s: string): Option<number> => {
+  const n = Number(s)
+  return s.trim() === ""
+    ? option.none
+    : Number.isNaN(n)
+    ? option.none
+    : option.some(n)
+}
+
+/**
+ * Takes a `bigint` and returns an `Option` of `number`.
+ *
+ * If the `bigint` is outside the safe integer range for JavaScript (`Number.MAX_SAFE_INTEGER`
+ * and `Number.MIN_SAFE_INTEGER`), it returns `option.none`. Otherwise, it converts the `bigint`
+ * to a number and returns `option.some(number)`.
```

Comment:
```suggestion
 * If the `bigint` is outside the safe integer range for JavaScript (`Number.MAX_SAFE_INTEGER`
 * and `Number.MIN_SAFE_INTEGER`), it returns `Option.none`. Otherwise, it converts the `bigint`
 * to a number and returns `Option.some(number)`.
```